### PR TITLE
Fix rabbitmq config decryption

### LINKED ISSUE #192499: rabbitmq-server doesn't load encrypted secret files anymore
### Describe the bug

`rabbitmq-server` doesn't load encrypted secret files anymore. It throws `config_decryption_error`.

```
22:27.853222+00:00 [error] <0.130.0> BOOT FAILED
22:27.853222+00:00 [error] <0.130.0> ===========
22:27.853222+00:00 [error] <0.130.0> Exception during startup:
22:27.853222+00:00 [error] <0.130.0>
22:27.853222+00:00 [error] <0.130.0> throw:{config_decryption_error,{key,default_pass},notsup}
22:27.853222+00:00 [error] <0.130.0>
22:27.853222+00:00 [error] <0.130.0>     rabbit_prelaunch_conf:decrypt_app/3, line 483
```

### Steps To Reproduce
1. Update to a recent version of Nixpkgs
2. Use an encrypted secret
3. `rabbitmq.service` fails to start repeatedly

### Expected behavior

No regression.

### Additional context

I'm bisecting.

<details><summary>Test and rebase log</summary>

```diff
commit ff34f03eda5a35e8c58486a145e7c8cb83966341 (hercules-ci/rabbitmq-config-decrypt-test, rabbitmq-config-decrypt-test)
Author: Robert Hensing <robert@roberthensing.nl>
Date:   Thu Sep 22 18:08:04 2022 +0100

    nixosTests.rabbitmq: Add config decryption and auth test

diff --git a/nixos/tests/rabbitmq.nix b/nixos/tests/rabbitmq.nix
index f8e8e61c47d..d0be85d559c 100644
--- a/nixos/tests/rabbitmq.nix
+++ b/nixos/tests/rabbitmq.nix
@@ -1,6 +1,12 @@
 # This test runs rabbitmq and checks if rabbitmq is up and running.
 
-import ./make-test-python.nix ({ pkgs, ... }: {
+import ./make-test-python.nix ({ pkgs, ... }:
+let
+  # in real life, you would keep this out of your repo and deploy it to a safe
+  # location using safe means.
+  configKeyPath = pkgs.writeText "fake-config-key" "hOjWzSEn2Z7cHzKOcf6i183O2NdjurSuoMDIIv01";
+in
+{
   name = "rabbitmq";
   meta = with pkgs.lib.maintainers; {
     maintainers = [ eelco offline ];
@@ -10,6 +16,27 @@ import ./make-test-python.nix ({ pkgs, ... }: {
     services.rabbitmq = {
       enable = true;
       managementPlugin.enable = true;
+
+      # To encrypt:
+      # rabbitmqctl --quiet encode --cipher blowfish_cfb64 --hash sha256 --iterations 10000 '<<"'"$RABBITMQ_PASSWORD_TEXT"'">>' "$RABB
ITMQ_CONFIG_KEY_TEXT"
+      config = ''
+        [ { rabbit
+          , [ {default_user, <<"alice">>}
+            , { default_pass
+              , {encrypted,<<"oKKxyTze9PYmsEfl6FG1MxIUhxY7WPQL7HBoMPRC/1ZOdOZbtr9+DxjWW3e1D5SL48n3D9QOsGD0cOgYG7Qdvb7Txrepw8w=">>}
+              }
+            , {config_entry_decoder
+              , [ {passphrase, {file, <<"${configKeyPath}">>}}
+                , {cipher, blowfish_cfb64}
+                , {hash, sha256}
+                , {iterations, 10000}
+                ]
+              }
+            % , {rabbitmq_management, [{path_prefix, "/_queues"}]}
+            ]
+          }
+        ].
+      '';
     };
     # Ensure there is sufficient extra disk space for rabbitmq to be happy
     virtualisation.diskSize = 1024;
@@ -23,5 +50,8 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         'su -s ${pkgs.runtimeShell} rabbitmq -c "rabbitmqctl status"'
     )
     machine.wait_for_open_port(15672)
+    machine.wait_until_succeeds(
+        '${pkgs.rabbitmq-java-client}/bin/PerfTest --time 10 --uri amqp://alice:dJT8isYu6t0Xb6u56rPglSj1vK51SlNVlXfwsRxw@localhost'
+    )
   '';
 })
```

```
git bisect start
# good: [4b05b28a81805cbb4fd3bab6ce6322041c3f240c] rabbitmq-server: 3.10.6 -> 3.10.7
git bisect good 4b05b28a81805cbb4fd3bab6ce6322041c3f240c
# bad: [cbb16e90e60fce2427002f5f319e715fd57f1a1b] Merge pull request #192250 from figsoda/changie
git bisect bad cbb16e90e60fce2427002f5f319e715fd57f1a1b
# good: [9985f0410014d620b0ea686cc15d05a5d7265abf] Merge pull request #188865 from r-ryantm/auto-update/lightspark
git bisect good 9985f0410014d620b0ea686cc15d05a5d7265abf
# good: [50bf8c918ba9b8c726daf6c6e18af31dd0b25461] Merge pull request #190778 from risicle/ris-trafficserver-9.1.3
git bisect good 50bf8c918ba9b8c726daf6c6e18af31dd0b25461
# bad: [00f1aa08ed98b694cb27d6eae6311363564a83bb] Merge pull request #190600 from Vonfry/update/fcitx5-20220910
git bisect bad 00f1aa08ed98b694cb27d6eae6311363564a83bb
# bad: [85a7a7147342919a85bee7acccf6ce0ae584ac09] Merge pull request #190966 from r-ryantm/auto-update/numix-icon-theme-square
git bisect bad 85a7a7147342919a85bee7acccf6ce0ae584ac09
# bad: [23022b2779ca1d06955069d00ddc349e0d7ca263] journaldriver: pin to openssl_1_1
git bisect bad 23022b2779ca1d06955069d00ddc349e0d7ca263
# bad: [5d26e8988df2451ede150b1106ef698294c41f10] Merge pull request #187562 from afh/master
git bisect bad 5d26e8988df2451ede150b1106ef698294c41f10
# skip: [5cc75bbb98cd5dc94be1dee3aba10547b62dbdd1] percona-xtrabackup_*: pin to openssl_1_1
### roberth: skipped because of mass rebuild, but next bisect point is also a mass rebuild
git bisect skip 5cc75bbb98cd5dc94be1dee3aba10547b62dbdd1

```

</details>

### Notify maintainers

@turion

### PR DESCRIPTION
**Pull request by roberth**
###### Description of changes
 
rabbitmq-server: Use openssl_1_1 to fix config decryption
    
Fixes #192499, nixosTests.rabbitmq
    
Reverts the effect of the openssl upgrade introduced in 0565276a075531b6dad21171de459473dd5e6a27 https://github.com/NixOS/nixpkgs/pull/150093

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by turion** [comment]
Does RabbitMQ not (yet) support OpenSSL 3.0? Would be nice if we can switch there eventually.

**Comment by roberth** [comment]
The RabbitMQ docs aren't explicit about the openssl dependency, but their Docker stuff is, and it's all OpenSSL 1.1. I agree.

File: nixos/tests/rabbitmq.nix [hunk 1]
```diff
@@ -23,5 +50,8 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         'su -s ${pkgs.runtimeShell} rabbitmq -c "rabbitmqctl status"'
     )
     machine.wait_for_open_port(15672)
+    machine.wait_until_succeeds(
+        '${pkgs.rabbitmq-java-client}/bin/PerfTest --time 10 --uri amqp://alice:dJT8isYu6t0Xb6u56rPglSj1vK51SlNVlXfwsRxw@localhost'
```
Comment (turion) [inline nixos/tests/rabbitmq.nix:54]: How did you generate `dJT8isYu6t0Xb6u56rPglSj1vK51SlNVlXfwsRxw`? How does it relate to password and config key and password?

Comment (turion) [reply nixos/tests/rabbitmq.nix:54]: I guess it's the password? Can you move it to the let block and give it a name?

**Review by turion — APPROVED** [review]
Just a minor doc nitpick, but I think this is great!

**Review by turion — APPROVED** [review]

**Pull request merged**